mathics/builtin/assignments/internals.py

```diff
@@ -153,7 +153,7 @@ def find_tag_and_check(lhs, tags, evaluation):
 
 
 def unroll_patterns(lhs, rhs, evaluation):
-    if type(lhs) is Symbol:
```

Comment:
Even if we hadn't added Constant, this idiom is deprecated. Probably for exactly the reason it needs to change here. When Constant was added, this code is now wrong as is, but had `isinstance()` been used in stead no change would have been needed.

Reply:
I just used this because the check is faster if `lhs` is not a `Symbol`.